File: android/app/src/main/java/com/privatecamera/PrivateCameraApp.kt
package com.privatecamera

import android.app.Application
import com.privatecamera.engine.DataPersistenceGuard

class PrivateCameraApp : Application() {
    override fun onCreate() {
        super.onCreate()
        // Verify data persistence on every launch
        DataPersistenceGuard.verifyOnLaunch(this)
    }
}

File: ios/PrivateCamera/App/PrivateCameraApp.swift
import SwiftUI

@main
struct PrivateCameraApp: App {
    @AppStorage("hasCompletedOnboarding") private var hasCompletedOnboarding = false

    init() {
        // Verify data persistence on every launch.
        // This ensures photo data survives app updates during development.
        DataPersistenceGuard.shared.verifyOnLaunch()
    }

    var body: some Scene {
        WindowGroup {
            if hasCompletedOnboarding {
                CameraView()
                    .preferredColorScheme(.dark)
            } else {
                OnboardingView()
                    .preferredColorScheme(.dark)
            }
        }
    }
}
